From a8d23bd73e89af2c0048c588799e7677a6a0c966 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Fri, 1 Aug 2025 11:55:26 +0000
Subject: [PATCH] search-provider: Quit after inactivity

Instead of keep running in the background, release the app after
10 seconds of inactivity.
---
 search-provider/lollypop-sp.in | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/search-provider/lollypop-sp.in b/search-provider/lollypop-sp.in
index 485c6a313..ea6e38994 100755
--- a/search-provider/lollypop-sp.in
+++ b/search-provider/lollypop-sp.in
@@ -155,7 +155,8 @@ class SearchLollypopService(Server, Gio.Application):
         Gio.Application.__init__(
                             self,
                             application_id='org.gnome.Lollypop.SearchProvider',
-                            flags=Gio.ApplicationFlags.IS_SERVICE)
+                            flags=Gio.ApplicationFlags.IS_SERVICE,
+                            inactivity_timeout=10000)
         self.cursors = {}
         self.task_helper = TaskHelper()
         self.settings = Settings.new()
@@ -188,6 +189,7 @@ class SearchLollypopService(Server, Gio.Application):
         return self.__search(terms)
 
     def GetResultMetas(self, ids):
+        self.hold()
         results = []
         try:
             for search_id in ids:
@@ -218,7 +220,7 @@ class SearchLollypopService(Server, Gio.Application):
                 results.append(d)
         except Exception as e:
             print("SearchLollypopService::GetResultMetas():", e)
-            return []
+        self.release()
         return results
 
     def GetSubsearchResultSet(self, previous_results, new_terms):
@@ -235,6 +237,7 @@ class SearchLollypopService(Server, Gio.Application):
         GLib.spawn_close_pid(pid)
 
     def __search(self, terms):
+        self.hold()
         ids = []
         search = noaccents(" ".join(terms))
         try:
@@ -250,12 +253,12 @@ class SearchLollypopService(Server, Gio.Application):
                 ids.append("t:"+str(track_id))
         except Exception as e:
             print("SearchLollypopService::__search():", e)
+        self.release()
         return ids
 
 def main():
     Gst.init(None)
     service = SearchLollypopService()
-    service.hold()
     service.run()
 
 if __name__ == '__main__':
-- 
GitLab

